test/verify/check-machines

```diff
@@ -483,6 +483,78 @@ class TestMachines(MachineCase):
 
         b.wait_in_text("#vm-subVmTest1-network-2-state span", "up")
 
+    def testVCPU(self):
+        b = self.browser
+        m = self.machine
+
+        self.startVm("subVmTest1")
+
+        self.login_and_go("/machines")
+        b.wait_in_text("body", "Virtual Machines")
+        b.wait_present("tbody tr th")
+        b.wait_in_text("tbody tr th", "subVmTest1")
+        b.click("tbody tr th") # click on the row header
+
+        b.wait_present("#vm-subVmTest1-state")
+        b.wait_in_text("#vm-subVmTest1-state", "running")
+        # Shut off VM for applying changes after save 
+        b.click("#vm-subVmTest1-off-caret")
+        b.wait_visible("#vm-subVmTest1-forceOff")
+        b.click("#vm-subVmTest1-forceOff")
+        b.wait_in_text("#vm-subVmTest1-state", "shut off")
+
+        b.wait_present("#vm-subVmTest1-vcpus-count") # wait for the tab
+        b.click("#vm-subVmTest1-vcpus-count") # open VCPU modal detail window
+
+        b.wait_present(".vcpu-detail-modal-table")
+        # Test basic vCPU properties
+        b.wait_val("#machines-vcpu-count-field", "1")
+        b.wait_val("#machines-vcpu-max-field", "1")
+
+        # Set new values
+        b.set_val("#machines-vcpu-max-field", "4")
+        b.set_val("#machines-vcpu-count-field", "3")
+
+        # Set new socket value
+        b.wait_present("#socketsSelect li[data-value=2] a")
+        b.click("#socketsSelect button");
+        b.click("#socketsSelect li[data-value=2] a")
+        b.wait_in_text("#socketsSelect button", "2")
+        b.wait_in_text("#coresSelect button", "1")
+        b.wait_in_text("#threadsSelect button", "2")
+
+        b.wait_not_present(".alert.alert-danger.dialog-error")
+
+        # Save
+        b.click(".apply")
+
+        # Check changes
+        b.wait_visible("#vm-subVmTest1-vcpus-count")
+        b.wait_in_text("#vm-subVmTest1-vcpus-count", "3")
+
+        # Check after boot
+        # Run VM
+        b.click("#vm-subVmTest1-run")
+        b.wait_in_text("#vm-subVmTest1-state", "running")
+
+        # Check VCPU count
+        b.wait_visible("#vm-subVmTest1-vcpus-count")
+        b.wait_in_text("#vm-subVmTest1-vcpus-count", "3")
```

Comment:
It seems you took out the `virsh dumpxml` check to make sure that the values actually made it through the UI. I think this was useful, and suggest putting it back. Even though it might be redundant, it's still a nice place to put this kind of information as a hint to developers.